instrumentation/log4j/log4j-2.13.2/library/src/main/java/io/opentelemetry/instrumentation/log4j/v2_13_2/OpenTelemetryLog4j.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.log4j.v2_13_2;
+
+import io.opentelemetry.sdk.logs.LogEmitter;
+import io.opentelemetry.sdk.logs.SdkLogEmitterProvider;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.concurrent.GuardedBy;
+
+public final class OpenTelemetryLog4j {
+
+  private static final Object LOCK = new Object();
+
+  @GuardedBy("LOCK")
+  private static LogEmitter logEmitter;
+
+  @GuardedBy("LOCK")
+  private static final List<OpenTelemetryAppender> APPENDERS = new ArrayList<>();
+
+  public static void initialize(SdkLogEmitterProvider sdkLogEmitterProvider) {
```

Comment:
How is this going to work as a javaagent instrumentation? In the agent we keep all the SDK classes hidden in the agent classloader, they're basically unavailable from the application classloader. The agent won't be able to register its emitterProvider in the application's `OpenTelemetryLog4j` class.
Was the idea about "log appender api" completely scrapped? What happened with that?

Reply:
We'll need it - https://github.com/open-telemetry/opentelemetry-java/issues/3807

Reply:
Took a stab at extracting out the appender API [here](https://github.com/open-telemetry/opentelemetry-java/pull/3880).